src/ui/compilationStatusView.ts

```diff
@@ -55,26 +60,37 @@ export default class CompilationStatusView {
     this.updateActiveDocumentStatus();
   }
 
+  private areParamsOutdated(params: ICompilationStatusParams): boolean {
+    const previous = this.documentStatusMessages.get(getVsDocumentPath(params));
+    return previous?.version !== undefined && params.version !== undefined
```

Comment:
```suggestion
    return previous?.version != null && params.version != null
```

nit: Although it's no longer problematic to compare against `undefined`, the code-base uses it interchangeably with `null` so far. See for example:

https://github.com/dafny-lang/ide-vscode/blob/d4ce528808aaaeaaf839ff11b8965f005887fef8/src/ui/dafnyVersionView.ts#L28

and

https://github.com/dafny-lang/ide-vscode/blob/2eec391d39ca71057c3ec92130d987ec4b5e8f72/src/tools/debounce.ts#L16-L19

But I'm open to refactorings if there is some need or preference to be more strict here.

Reply:
Oh I did not see it would be accepted by the linter. Nice. I used your expression.